wallet/src/main/scala/org/bitcoins/wallet/models/AddressRecord.scala

```diff
@@ -0,0 +1,135 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.hd.{
+  HDChainType,
+  HDCoinType,
+  HDPurpose,
+  HDPurposes,
+  LegacyHDPath,
+  NestedSegWitHDPath,
+  SegWitHDPath
+}
+import org.bitcoins.core.protocol.{
+  Bech32Address,
+  BitcoinAddress,
+  P2PKHAddress,
+  P2SHAddress
+}
+import org.bitcoins.core.protocol.script.{ScriptPubKey, ScriptWitness}
+import org.bitcoins.crypto.{ECPublicKey, Sha256Hash160Digest}
+
+case class AddressRecord(
+    purpose: HDPurpose,
+    accountCoin: HDCoinType,
+    accountIndex: Int,
+    accountChain: HDChainType,
+    addressIndex: Int,
+    address: BitcoinAddress,
+    pubKey: ECPublicKey,
```

Comment:
How does this handle scripts with multiple public keys?

Reply:
This PR is refactoring PR so there will be no new features here.